### PR TITLE
Feature/add gke tests

### LINKED ISSUE #515: Add tests for on GKE end-to-end example
## Story
Add automated tests for the on GKE end-to-end example.

## Proposed Solution
Assert the cluster is running, is regional, uses public nodes, has expected node count, and check that all pods are healthy. Test client can connect to server and test cron job.

### PR DESCRIPTION
**Pull request by gkowalski-google**
Resolves #515

### CODE REVIEW

File: modules/on_gke/variables.tf [hunk 1]
```diff
@@ -911,7 +911,7 @@ variable "git_sync_wait" {
 
 variable "helm_chart_version" {
   description = "The version of the Helm chart to use"
-  default     = "2.1.0"
+  default     = "2.2.0-rc1"
```
Comment (gkowalski-google) [inline modules/on_gke/variables.tf]: Currently using this version until 2.2.0 is released.

File: test/integration/on_gke_end_to_end/controls/kubectl.rb [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+      end
+
+      describe "tiller-deploy" do
+        let(:pod) do
+          all_pods.select { |n| n.metadata.labels.name == "tiller" }.first
+        end
+
+        it "has the expected status" do
+            expect(pod.status.phase).to eq 'Running'
+        end
```
Comment (gkowalski-google) [inline test/integration/on_gke_end_to_end/controls/kubectl.rb]: These tests will ensure the minimum Forseti pods are running.

File: examples/on_gke_end_to_end/outputs.tf [hunk 3]
```diff
@@ -14,6 +14,22 @@
  * limitations under the License.
  */
 
+output "client_token" {
+  description = "The bearer token for auth"
+  sensitive   = true
+  value       = base64encode(data.google_client_config.default.access_token)
+}
+
+output "ca_certificate" {
```
Comment (kevensen) [inline examples/on_gke_end_to_end/outputs.tf]: Is this needed for testing?

Comment (gkowalski-google) [reply examples/on_gke_end_to_end/outputs.tf]: yes

File: modules/on_gke/main.tf [hunk 4]
```diff
@@ -69,7 +69,7 @@ locals {
     "storage-api.googleapis.com",
     "groupssettings.googleapis.com",
   ]
-  workload_identity                = "${var.project_id}.svc.id.goog"
+  workload_identity_namespace      = var.workload_identity_namespace == null ? "${var.project_id}.svc.id.goog" : var.workload_identity_namespace
```
Comment (kevensen) [inline modules/on_gke/main.tf]: I recommend not making this conditional: input variable or bust.

**Comment by gkowalski-google** [comment]
Due to [this bug with Helm](https://github.com/helm/helm/issues/6361), these tests will not pass and will need to upgrade to Helm 3 to get the fix.

Related to #548

**Pull request closed without merging**